Update documentation on linked release URLs

yolile opened the pull request:
closes #844

yolile committed:
Update documentation on linked release URLs

Signed-off-by: Yohanna Lisnichuk <yohanitalisnichuk@gmail.com>

jpmckinney commented:
Hmm, the old, surrounding text is not very clear. A short JSON snippet can demonstrate the overall structure of the record. There can be a better description of the two options (embedded releases and linked releases), and perhaps an explanation of their different benefits. The format of linked releases can be better described (perhaps with a short jsonschema directive of the three fields). A discussion of how linked releases refer to release packages can follow.

Do you want to update that entire section?

yolile commented:
@jpmckinney sure! I will update the record_reference.md file with your suggestions, but should I leave record-package-schema.json as it is now? (with this PR changes)

jpmckinney commented:
Let's update the Markdown first. It might give us some ideas for improving the schema description. Please also add a changelog entry (you can put it under the Schema heading, even though it also changes the Markdown).

yolile committed:
Merge branch '1.1.5-dev' of github.com:open-contracting/standard into update-linked-releases-url-doc

yolile committed:
[WIP] Update releases in record reference guidance

Signed-off-by: Yohanna Lisnichuk <yohanitalisnichuk@gmail.com>

yolile committed:
Update documentation about records releases

Signed-off-by: Yohanna Lisnichuk <yohanitalisnichuk@gmail.com>

yolile commented:
@jpmckinney maybe something like this?

jpmckinney commented:
Yes, looks great! Let me know when ready for review.

yolile committed:
Update records reference doc

Signed-off-by: Yohanna Lisnichuk <yohanitalisnichuk@gmail.com>

yolile commented:
@jpmckinney ready for review

jpmckinney committed:
Fix typo in filename

jpmckinney committed:
Rewrite records reference

- Add note that schema takes precedence, as on release schema page
- Avoid use of the word "copy", which implies OCDS data that exists prior to the record's creation
- Add section with considerations about embedded releases versus linked releases
- Display linked release schema snippet
- Remove link to issue to be closed
- Reduce repetition

jpmckinney commented:
@yolile Can you review my new changes this week? I'd like to merge and announce the 1.1.5 draft before I take vacation next week.

Link: https://standard.open-contracting.org/update-linked-releases-url-doc/en/schema/records_reference/

jpmckinney committed:
Fix missing anchor character

jpmckinney committed:
Merge branch '1.1.5-dev' into update-linked-releases-url-doc

jpmckinney committed:
Remove link to release package metadata section, and instead repeat the two paragraphs

jpmckinney committed:
Update changelog

yolile commented on docs/schema/records_reference.md:
```diff
@@ -28,81 +13,112 @@ The versioned release format is designed to support analysis of how data has cha
+### Versioned release
 
-Third parties can use the information in the releases list to fetch source data, compile and verify their own version history for a contract.
+A versioned release contains the history of all the data in the compiled release, including which fields were changed, when they were changed, and the release in which they were changed.
```
maybe add "..including the fields first value, which fields were changed.." as the versionedRelease doesn't include only the fields that changed but all of them

jpmckinney replied:
I changed the paragraph to:

> A versioned release aggregates all values of all fields from all releases. For each field, it describes the history of values from the initial value to the current value, including the `id`, `date` and `tag` of the release in which it was changed.

yolile replied:
perfect!

yolile commented on docs/schema/records_reference.md:
```diff
@@ -28,81 +13,112 @@ The versioned release format is designed to support analysis of how data has cha
+
+#### Embedded releases
+
+An embedded release follows the [release schema](reference).
```
@jpmckinney maybe add "So instead of just linking the release, here the whole releases are embedded" or similar, to describe more what an embedded release is

jpmckinney replied:
Good point. I'll use:

> In other words, instead of linking to a release within a release package as above, that release is entirely included in the record.

yolile commented on docs/schema/records_reference.md:
````diff
@@ -28,81 +13,112 @@ The versioned release format is designed to support analysis of how data has cha
+
+A linked release follows a simple schema:
+
+```eval_rst
````
@jpmckinney the first row of the table is missing the title/description 
![image](https://user-images.githubusercontent.com/6834122/84403250-7bb30380-abd3-11ea-839b-e5a3a0588c66.png)

jpmckinney replied:
Thanks. I'll fix the JSON Pointer so that the table starts at the `url` row and omits the `/0`.

yolile commented on docs/schema/records_reference.md:
```diff
@@ -28,81 +13,112 @@ The versioned release format is designed to support analysis of how data has cha
-* Releases (required)
-* Compiled Release (recommended)
-* Versioned Release (optional)
+A record **may** contain a [versionedRelease](#versioned-release) object, which aggregates all values from all releases in a single object. The versioned release is designed to make it easy to see how values change from one release to another, and will often be generated by data users, rather than by publishers.
```
"all values of all fields"?

jpmckinney replied:
Will do.

jpmckinney committed:
Improvements to records reference page

- Improve descriptions of embedded releases and versioned releases
- Fix JSON Schema table for linked releases

jpmckinney commented:
@yolile I've added a commit with the changes. Ready for review :)

yolile commented:
@jpmckinney all good and clear! (I cannot approve as I'm the PR's author)
